texasfortytwo.game.php

```diff
@@ -1,499 +1,502 @@
-		// Important: Must only return game state visible to this player!
-		$current_player_id = self::getCurrentPlayerId();
-    $result = array();
-
-    // TODO(isherman): Do we need to return player scores at all? I doubt our
-		// custom game logic cares about it, and I bet this is returned separately
-		// for the pre-canned BGA UI surfaces.
-		// Publicly visible state about the players.
-    $result['players'] = self::getCollectionFromDb(
-			  'SELECT player_id id, player_score score FROM player');
-
-    // Dominoes in the current player's hand.
-		$result['hand'] = $this->getDominoesInLocation('hand', $current_player_id);
-		// Dominoes in play on the table.
-    $result['table'] = $this->getDominoesInLocation('table');
-    return $result;
-  }
-
-    /*
-        getGameProgression:
-
-        Compute and return the current game progression.
-        The number returned must be an integer beween 0 (=the game just started) and
-        100 (= the game is finished or almost finished).
-
-        This method is called each time we are in a game state with the "updateGameProgression" property set to true
-        (see states.inc.php)
-    */
-    function getGameProgression()
-    {
-        // TODO: compute and return the game progression
-
-        return 0;
+    ]);
+  }
+
+  protected function getGameName() {
+    // Used for translations and stuff. Please do not modify.
+    return 'texasfortytwo';
+  }
+
+  // Called once, when a new game is launched. Initializes game state.
+  protected function setupNewGame($players, $options = []) {
+    self::initializePlayers($players);
+
+    // Initialize game state.
+    // TODO(isherman): Call self::setGameStateInitialValue for any relevant
+    // game state variables here.
+
+    // Initialize game statistics. Must match the list of stats defined in
+    // stats.inc.php.
+    // TODO(isherman): Call self::initStat for all defined statistics.
+
+    self::initializeDeck();
+
+    // Begin the game by activating the first player.
+    $this->activeNextPlayer();
+  }
+
+  // Inserts a set of fields into the database named `$db_name`;
+  // `$fields`: an array of field names, e.g. 'player_id'.
+  // `$rows`: an array of arrays, where each inner array defines the values for
+  //     one row, specified in the same order as
+  //     `$field_names`.
+  private static function insertIntoDatabase($db_name, $fields, $rows) {
+    $to_sql_row = function ($row) {
+      return "('".join("','", $row)."')";
+    };
+    $fields = join(',', $fields);
+    $values = join(',', array_map($to_sql_row, $rows));
+    self::DbQuery("INSERT INTO $db_name ($fields) VALUES $values");
+  }
+
+  // Initializes the player database for the game. Called once, when a new game
+  // is launched.
+  private function initializePlayers($players) {
+    $fields = [
+            'player_id',
+            'player_color',
+            'player_canal',
+            'player_name',
+            'player_avatar',
+        ];
```

Comment:
nit: The formatting here looks weird – is that just because it was weird before?